android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/NavGraph.kt

```diff
@@ -56,13 +58,25 @@ fun NavGraphBuilder.mainGraph(navController: NavController) {
                 navController.popBackStack()
             },
             onStartSurveyClick = {
+                navController.currentBackStackEntry?.arguments?.putParcelable(
+                    Argument.surveyQuestions,
+                    it
+                )
                 navController.navigate(AppDestination.SurveyQuestions)
             }
         )
     }
     composable(AppDestination.SurveyQuestions) {
-        SurveyQuestionsScreen(onCloseClick = {
-            navController.popBackStack()
-        })
+        val arguments = navController.previousBackStackEntry?.arguments
+        val surveyQuestionsArgument =
+            arguments?.getParcelable<SurveyQuestionsArgument>(Argument.surveyQuestions)
+        val viewModel: SurveyQuestionsViewModel = getViewModel(
+            parameters = { parametersOf(surveyQuestionsArgument) }
+        )
+        SurveyQuestionsScreen(
+            viewModel,
+            onCloseClick = {
+                navController.popBackStack()
+            })
```

Comment:
I think we could replace by `onCloseClick = navController::popBackStack`

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/115/commits/194ae2915a117e34beaee3538b87ceae2f7e31ac